### ISSUE TITLE
Columbia (Country)

### ISSUE THREAD

**Issue by msirenko** [openaddresses/openaddresses#2679]
Was chatting with a partner of ours about the structure of columbian addresses. It seems like Ecuador may follow a similar pattern.

**Tl;Dr**:  looks like addresses in Columbia are all calculated, using the distance from the nearest intersections. To me this sounds like the country is 100% interpolated.  Is this useful / actionable to OpenAddresses?

cc @migurski @ingalls @iandees 

Full description below

-------

A city address in Colombia conforms to the following example:

Calle 150 A # 101 - 20 (http://www.openstreetmap.org/?mlat=4.74839&mlon=-74.09076#map=19/4.74839/-74.09076)

Is composed of:

 * Closest street of the point, in this case is Calle 150 A : Composed of the type of Street a number and letters
 * 101 corresponds to the street of the intersection closest to the origin of the city (0,0), near the point.  It can have optionally letters, and it's named in OSM completely. Pointing the intersection http://www.openstreetmap.org/?mlat=4.74820&mlon=-74.09037#map=19/4.74820/-74.09037
 * The last one is the distance in meters to the intersection of the two streets. It's even or odd depending on the side of the closest street.

We have different types(names) for streets, at the beginning we have Calle and Carrera and the others are circular, transversal, diagonal, paralela, but they follow the same conventions.

Calle number increases to the north of the origin of the city and Carrera number increases to the west of the origin of the city.

I personally wrote a starting point for geocoding, you can find the source code at https://github.com/ikks/osmhelpertools/ based on osm data that goes close to the street intersection, but did not calculate the third coordinate, the distance along the line could be calculated and assigning even or odd would be the missing main bits to take into account.  You can look at this system in action in http://test.openstreetmap.co/

We have also other constraints, as you can see intersection of calle 1 and carrera 1 has carreras to the east and calles to the south, they are properly labeled usually with the letter E for East and the letter S for south accordingly.

**Comment by msirenko** [comment]
![image](https://cloud.githubusercontent.com/assets/7937582/24083877/9510bd22-0cad-11e7-9bd0-bf8887797d94.png)
![image](https://cloud.githubusercontent.com/assets/7937582/24083879/a2607b98-0cad-11e7-95a5-20c47ccab9ba.png)

**Comment by thatdatabaseguy** [comment]
Hi Max - Al here, did some of the work on Colombia for OpenAddresses in order to obtain some better Colombian training data for the address parser in [libpostal](https://github.com/openvenues/libpostal).

Though the meters-from-the-intersection piece seems strange at first, AFAICT it's essentially just a house number. From a US perspective it wouldn't be too dissimilar from having a "300 block" between 3rd and 4th Street, but instead of the house numbers being sequential, they're based on how far the building is from the intersection. Most buildings in cities like Bogotá are labeled with the compound number (e.g. https://goo.gl/maps/YeB9b2ckJK32)

In the two Colombia data sets currently in OpenAddresses, the lat/lons are correctly geocoded to the buildings not the intersections. The fields listed in said data sets are similar to those above e.g. STREET="CR 24C", NUMBER="95-38".

So probably not pertinent to OpenAddresses, but useful for people adding addresses to OSM (as well as for navigation and geocoding in those cities).
